doc/nrf/ug_partition_manager.rst

```diff
@@ -0,0 +1,203 @@
+   and placement is implied by the size and placement of the sub image
+   partitions.
+
+.. _pm_yaml_format:
+
+Configuration file format
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The format of the :file:`pm.yml` file is as follows:
+
+.. code-block:: yaml
+
+   partition_name:
+      option_dict_name:
+         option_specific_values
+
+
+Where ``option_dict_name`` can be:
+
+placement: dict{}
+   The placement of the partition relative to other partitions, the end of flash, or the root image partition `app`.
+   The placement is formatted as a yaml dict.
+   The valid keywords are listed below.
+
+      before: list
+         Place the partition before the first existing partition in the list.
+
+      after: list
+         Place the partition after the first existing partition in the list.
+
+   The valid values in the lists are `app`, `end`, or the name of any partition.
+   It is not possible to place the partition after `end`.
+
+sub_partitions: list
+   Define sub partitions which span across one or more other partitions.
+   The sub partitions split the total size of the spanned partitions, and therefore have equal size.
+   This property cannot be used together with the `placement` property.
+   When this property is set for a partition, it is required that the `span` property is also set.
+
+span: list
+   When creating sub_partitions, this option lists what partitions the sub_partitions should span across.
+   This property cannot be used together with the `placement` property.
+
+.. _pm_yaml_preprocessing:
+
+Configuration file preprocessing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each :file:`pm.yml` file is preprocessed.
+Example of preprocessing is shown below:
+
+.. code-block:: yaml
+
+   #include <autoconf.h>
+
+   # 'b0' is the name of the image partition.
+   b0:
+
+     # b0 is placed before the mcuboot partition if the mcuboot partition
+     # exists, otherwise it is stored before the app partition.
+     placement: {before: [mcuboot, app]}
+
+   # Don't define the provision partition if the SoC is nRF9160, this because
+   # the provisioning data will be stored in the UICR->OTP data region.
+
+   #ifndef CONFIG_SOC_NRF9160
+
+   # 'provision' is the name of the placeholder partition.
+   provision:
+     # This partition is stored at the very end of flash.
+     placement: last
+
+   #endif /* CONFIG_SOC_NRF9160 */
+
+.. _pm_yaml_partition_types:
+
+Partition types
+~~~~~~~~~~~~~~~
+
+It is required that each :file:`pm.yml` defines exactly one *image partition*.
+This is done by using the same name for the partition as the image name.
+
+All other partitions which have the `placement` property set are *placeholder
+partitions*.
+
+Partitions which have the `sub_partitions` property set are *phony partitions*,
+and do not occupy space in flash.
+
+In addition to the configuration provided in :file:`pm.yml`, each partition with the `placement` property set must have a corresponding size configuration in the image's ``Kconfig`` file.
```

Comment:
What is an "image's `Kconfig` file"? If an image is a zephyr application, it will have many Kconfig files, some in the application itself, and some in the zephyr tree. Does this just mean that the `CONFIG_PM_PARTITION_SIZE_[PARTITION_NAME]` Kconfig option must be set for each image? If so, might be worth saying so.

If you move the partition type docs into the configuration format section above, I think you can split this into its own section named "Configuring partition sizes" or so.

Reply:
This will be changed in the future as the Kconfig is no longer provided to pm.py, the preprocessing will be used instead. So this comment no longer applies.